src/libraries/icubmod/bcbBattery/bcbBattery.cpp

```diff
@@ -285,26 +204,27 @@ void BcbBattery::run()
         sprintf(buff, "battery status: %+6.1fA   % 6.1fV   charge:% 6.1f%%    time: %s", battery_current, battery_voltage, battery_charge, battery_timestamp);
         yDebug("BcbBattery::run() log_buffer is: %s", buff);
     }
+
+    //empty the buffer
+    serial_buff[0] = 0;
+    pSerial->flush();
 }
 
 bool BcbBattery::getBatteryVoltage(double &voltage)
 {
-    lock_guard<mutex> lck(mtx);
-    voltage = battery_voltage;
+    voltage = batteryReader->battery_voltage;
```

Comment:
Even if this is just a read, it is still possible that there is a concurrent write on the ``run`` method. I guess that these variables need to be [``atomic``](https://en.cppreference.com/w/cpp/atomic/atomic) if you don't want to use any mutex.